smartsim/_core/mli/comm/channel/dragonchannel.py

```diff
@@ -43,24 +43,59 @@
 class DragonCommChannel(cch.CommChannelBase):
     """Passes messages by writing to a Dragon channel"""
 
-    def __init__(self, key: bytes) -> None:
-        """Initialize the DragonCommChannel instance"""
-        super().__init__(key)
-        self._channel: dch.Channel = dch.Channel.attach(key)
+    def __init__(self, channel: "dch.Channel", recv_timeout: int = 0) -> None:
+        """Initialize the DragonCommChannel instance
+
+        :param channel: a channel to use for communications
+        :param recv_timeout: a default timeout to apply to receive calls"""
+        serialized_ch = channel.serialize()
+        safe_descriptor = base64.b64encode(serialized_ch).decode("utf-8")
+        super().__init__(safe_descriptor)
+        self._channel = channel
+        self._recv_timeout = recv_timeout
+
```

Comment:
Really just curious why we're moving away form attaching with a key? Although maybe I just haven't gotten to that part of the code yet!

Reply:
Disregard! I see that that logic was moved to the `from_descriptor` method.

Reply:
The factory method `from_descriptor` is for re-creating a `CommChannel` from an existing descriptor. Having this constructor lets us do the initial construction of the object. 

From another perspective, we can just see that the way it was repeated the same thing for the constructor and `from_descriptor`. This keeps it DRY